packages/client/src/runtime/core/types/GetResult.ts

```diff
@@ -48,21 +48,19 @@ export type FluentOperation =
 type Count<O> = { [K in keyof O]: Count<number> } & {}
 
 // prettier-ignore
-export type GetFindResult<P extends Payload, A> =
-  {} extends A ? DefaultSelection<P> :
-  A extends 
-  | { select: infer S } & Record<string, unknown>
-  | { include: infer S } & Record<string, unknown>
-  ? S extends undefined ? DefaultSelection<P> :
-    {
-      [K in keyof S as S[K] extends false | undefined | null ? never : K]:
-        S[K] extends object
+export type GetFindResult<P extends Payload, _A, A = { 0: _A, 1: { select: undefined } }[Equals<_A, any>]> =
```

Comment:
`{ 0: _A, 1: { select: undefined } }[Equals<_A, any>]` this means if `_A` equals `any` then return `{ select: undefined }` otherwise return `_A`.

Reply:
fyi, this is more a perf improvement, so I moved it out of the PR to https://github.com/prisma/prisma/pull/20161

Reply:
(I left the two utilities though)